keyleak/archive_scanner.py

```
@@ -0,0 +1,110 @@
+        return dest
+
+    raise ArchiveScanError(f"Unsupported archive format: {archive_path}")
+
+
+def _safe_extract_zip(zf: zipfile.ZipFile, dest: Path) -> None:
+    dest = dest.resolve()
+    for info in zf.infolist():
+        target = (dest / info.filename).resolve()
+        if not str(target).startswith(str(dest)):
+            raise ArchiveScanError(f"Path traversal in zip entry: {info.filename}")
+    zf.extractall(dest)
```

Ruff (0.15.14)

[error] Uses of `tarfile.extractall()`

(S202)